packages/patternfly-4/react-core/src/experimental/components/OverflowMenu/OverflowMenu.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/OverflowMenu/overflow-menu';
+import { css, getModifier } from '@patternfly/react-styles';
+
+export interface OverflowMenuProps extends React.HTMLProps<HTMLDivElement> {
+  /** Any elements that can be rendered in the menu */
+  children?: any;
+  /** Additional classes added to the OverflowMenu. */
+  className?: string;
+  /** Indicates breakpoint at which to switch between horizontal menu and vertical dropdown */
+  breakpoint: 'Md' | 'Lg' | 'Xl';
+}
+
+export const OverflowMenu: React.SFC<OverflowMenuProps> = ({
+  className,
+  breakpoint,
+  children,
+  ...props
+}) => (
+  <div
+    {...props}
+    className={css(
+      styles.overflowMenu,
+      getModifier(styles.modifiers, `showOn${breakpoint}`),
+      className
+    )}
+  >
+    {React.Children.map(children, component =>
+      React.cloneElement(component, {
```

Comment:
i've often ran into issues with cloneElement in the past. This could break for example it looks like currently the structure it expects is:
```
<OverflowMenu>
  <OverflowMenuContent>...</OverflowMenuContent>
  <OverflowMenuControl>...</OverflowMenuControl>
<OverflowMenu>
```
But if user did this:
```
const MyMenuContent = (
  <OverflowMenuContent>...</OverflowMenuContent>
);
return (
<OverflowMenu>
  {MyMenuContent}
  <OverflowMenuControl>...</OverflowMenuControl>
<OverflowMenu>
)
```
it would clone on MyMenuContent and not on  OverflowMenuContent.
My recommendation would be to not use cloneElement here and throughout if avoidable, and you react context instead.

Reply:
Thanks - removed cloneElement and replaced with Context throughout the components.